_posts/2025-03-31-zscore-hybrid-search.md

```diff
@@ -0,0 +1,170 @@
+meta_description: Learn about z score normalization using the Neural Search plugin in OpenSearch 3.0-beta. Discover how this new approach to hybrid search merges results from multiple query sources for improved relevance.
+---
+
+In the world of search engines and machine learning, data normalization plays a crucial role in ensuring fair and accurate comparisons between different features or scores. 
+Hybrid query uses multiple normalization techniques for preparing final results, main two types are score based normalization and rank base combination. In score base normalization, min-max normalization doesn’t work well with outliers (Outliers are data points that significantly differ from other observations in a dataset. 
+In the context of normalization techniques like Min-Max scaling and Z-score (Standard Score) normalization, outliers can have a substantial impact on the results). In this blogpost we would introduce another normalization technique called as z-score which was added in OpenSearch 3.0.0-beta release. 
+Let's dive into what Z-score normalization is, why it's important, and how it's being used in OpenSearch.
+
+## What is Z-Score Normalization?
+
+Z-score normalization, also known as standardization, is a method of scaling data using mean and standard deviation. The formula for calculating the Z-score is:
+Z = (X - μ) / σ
+Where:
+
+* X is the original value
+* μ is the mean of the population
+* σ is the standard deviation of the population
+
+## When to use Z Score?
+
+Considering your index’s structure can help you decide which one to choose since each has advantages of its own. If your documents are more similar to one another and the top-k results of a typical query return documents that are very similar to one another and clustered together within the index, as seen in the graph below, Min-Max may be a better option.
+
+![Image for min max distribution](/assets/media/blog-images/2025-03-31-zscore-hybrid-search/blogpost1.jpg){: .img-fluid}
+
+However, Z-Score is more suited if the results are more evenly distributed and have some characteristics of a normal distribution, as shown in the example below.
+
+![Image for zscore distribution](/assets/media/blog-images/2025-03-31-zscore-hybrid-search/blogpost2.jpg){: .img-fluid}
+
+The basic flow to use between min max and z score looks like below:
+
+![Image for flow](/assets/media/blog-images/2025-03-31-zscore-hybrid-search/blogpost3.png){: .img-fluid}
+
+### How to use Z Score?
+
+To use z_score, create a search pipeline and specify `z_score` as the `technique`:
+
+```
+PUT /_search/pipeline/z_score-pipeline
+{
+    "description": "Zscore processor for hybrid search",
+    "phase_results_processors": [
+        {
+            "normalization-processor": {
+                "normalization": {
+                    "technique": "z_score"
+                },
+                "combination": {
+                    "technique": "arithmetic_mean"
```

Comment:
I would give an example without combination technique, we support only one AM, this may give false impression that we support same subset of three techniques, as for other normalization methods.